scheduler/orchestrator/anomaly_detector.py

```
@@ -157,227 +167,258 @@ def format_report(self) -> str:
-        cmd = f"aws s3 ls s3://{self.s3_bucket}/{hub}/{stage}/ --profile {self.aws_profile}"
-        result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
-        
+        cmd = [
+            "aws",
+            "s3",
+            "ls",
+            f"s3://{self.s3_bucket}/{hub}/{stage}/",
+            "--profile",
+            "dpla",
+        ]
+        result = subprocess.run(cmd, capture_output=True, text=True)
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)